test/cpp/interop/backend_metrics_lb_policy_test.cc

```diff
@@ -0,0 +1,143 @@
+#include "test/core/util/port.h"
+#include "test/core/util/test_config.h"
+#include "test/cpp/interop/backend_metrics_lb_policy.h"
+
+namespace grpc {
+namespace testing {
+namespace {
+
+class EchoServiceImpl : public grpc::testing::TestService::CallbackService {
+ public:
+  grpc::ServerUnaryReactor* UnaryCall(
+      grpc::CallbackServerContext* context,
+      const grpc::testing::SimpleRequest* /* request */,
+      grpc::testing::SimpleResponse* /* response */) override {
+    auto reactor = context->DefaultReactor();
+    reactor->Finish(grpc::Status::OK);
+    return reactor;
+  }
+};
+
+class Server {
+ public:
+  Server() : port_(grpc_pick_unused_port_or_die()) {
+    server_thread_ = std::thread(ServerLoop, this);
+    grpc_core::MutexLock lock(&mu_);
+    cond_.WaitWithTimeout(&mu_, absl::Seconds(1));
+  }
+
+  ~Server() {
+    server_->Shutdown();
+    server_thread_.join();
+  }
+
+  std::string address() const { return absl::StrCat("localhost:", port_); }
+
+ private:
+  static void ServerLoop(Server* server) { server->Run(); }
+
+  void Run() {
+    ServerBuilder builder;
+    EchoServiceImpl service;
+    auto server_metric_recorder =
+        grpc::experimental::ServerMetricRecorder::Create();
+    server_metric_recorder->SetCpuUtilization(.5f);
+    grpc::experimental::OrcaService orca_service(
+        server_metric_recorder.get(),
+        grpc::experimental::OrcaService::Options().set_min_report_duration(
+            absl::Seconds(1)));
+    builder.RegisterService(&orca_service);
+    builder.RegisterService(&service);
+    builder.AddListeningPort(address(), InsecureServerCredentials());
+    auto grpc_server = builder.BuildAndStart();
+    server_ = grpc_server.get();
+    {
+      grpc_core::MutexLock lock(&mu_);
+      cond_.SignalAll();
+    }
+    grpc_server->Wait();
+  }
+
+  int port_;
+  grpc_core::Mutex mu_;
+  grpc_core::CondVar cond_;
+  std::thread server_thread_;
+  grpc::Server* server_;
+};
+
+TEST(BackendMetricsLbPolicyTest, TestOobMetricsReceipt) {
+  LoadReportTracker tracker;
+  grpc_core::CoreConfiguration::RegisterBuilder(RegisterBackendMetricsLbPolicy);
+  Server server;
+  ChannelArguments args = tracker.GetChannelArguments();
+  args.SetLoadBalancingPolicyName("test_backend_metrics_load_balancer");
+  auto channel = grpc::CreateCustomChannel(server.address(),
+                                           InsecureChannelCredentials(), args);
+  auto stub = grpc::testing::TestService::Stub(channel);
+  ClientContext ctx;
+  SimpleRequest req;
+  SimpleResponse res;
+  ASSERT_EQ(stub.UnaryCall(&ctx, req, &res).error_code(), grpc::OK);
+  // This report is sent on start, available immediately
+  auto report = tracker.WaitForOobLoadReport(
+      [](auto report) { return report.cpu_utilization() == 0.5; },
+      absl::Milliseconds(1), 3);
+  ASSERT_TRUE(report.has_value());
+  EXPECT_EQ(report->cpu_utilization(), 0.5);
+  for (size_t i = 0; i < 3; i++) {
+    // Wait for slightly more that 1 min
```

Comment:
This should be `// Wait for slightly more than 1 second` ?

Reply:
Done. Thanks for pointing out!